### PR TITLE
STYLE: upgrade flake8 to version 4+

### LINKED ISSUE #45051: STYLE upgrade flake8 to version 4+
flake8 isn't getting updated by the auto-update job because the url

https://github.com/pandas-dev/pandas/blob/71312683b41b5177faf7ecd63555059504853cbd/.pre-commit-config.yaml#L38-L39

has changed. flake8 is now hosted at https://github.com/PyCQA/flake8 .

Task here is to change the repo url, set the latest flake8 version, and fixup any errors which result from
```
pre-commit run flake8 --all-files
```

### PR DESCRIPTION
**Pull request by MiTo0o**
- [x] closes #45051 
- [x] tests passed
- [x] Ensure all linting tests pass, see [here](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#pre-commit) for how to run them
- [x] bumping version of flake8 and update repo url

### CODE REVIEW

**Comment by MiTo0o** [comment]
Sorry I'm unfamiliar with the  `pre-commit run --show-diff-on-failure --color=always --all-files` command, which I'm failing. It even fails when i run it locally on the master branch. Can someone explain it to me? Sorry, I'm new to this.

File: .pre-commit-config.yaml [hunk 1]
```diff
@@ -98,7 +98,7 @@ repos:
         entry: flake8-rst
         types: [rst]
         args: [--filename=*.rst]
-        additional_dependencies: [flake8-rst==0.7.0, flake8==3.7.9]
+        additional_dependencies: [flake8-rst==0.7.0, flake8==4.0.1]
```
Comment (MarcoGorelli) [inline .pre-commit-config.yaml:101]: you don't need to upgrade this one

**Review by MarcoGorelli — CHANGES_REQUESTED** [review]

File: .pre-commit-config.yaml (hunk 1 above)
Comment (MarcoGorelli) [reply .pre-commit-config.yaml:101]: this is just a dependency for `flake8-rst`, which seems to be broken for modern versions of flake8

we should probably replace it, but for now it's OK to keep it with flake8 pinned

Comment (MiTo0o) [reply .pre-commit-config.yaml]: Ok, sounds good. I reverted it.

File: .pre-commit-config.yaml [hunk 2]
```diff
@@ -35,13 +35,13 @@ repos:
...
     hooks:
     -   id: flake8
         additional_dependencies: &flake8_dependencies
-        - flake8==3.9.2
-        - flake8-comprehensions==3.1.0
+        - flake8==4.0.1
+        - flake8-comprehensions==3.7.0
         - flake8-bugbear==21.3.2
         - pandas-dev-flaker==0.2.0
     -   id: flake8
```
Comment (MarcoGorelli) [inline .pre-commit-config.yaml:47]: looks like flake8 v4 no longer works on Cython files (and that if it did before, it was purely by accident) https://github.com/PyCQA/flake8/issues/1482

@jbrockmendel how important is flake8 linting of cython files? Is it OK to remove it, or would you rather have a separate hook for cython linting which pins flake8 at 3.9.2?

Comment (jbrockmendel) [reply .pre-commit-config.yaml:47]: id be OK with not having it; i get false-positives about "cimport" all the time

Comment (MarcoGorelli) [reply .pre-commit-config.yaml:47]: sure, it's a hack anyway - @MiTo0o can you remove these cython flake8 checks too please, as well as their config files?

Comment (MiTo0o) [reply .pre-commit-config.yaml]: Sounds good, I just pushed some changes, lets see if I pass everything and if it meets your expectations.

**Review by MarcoGorelli — APPROVED** [review]
Looks good to me, thanks!

**Pull request merged**